node/bft-consensus/src/state.rs

```diff
@@ -130,8 +130,16 @@ impl<N: Network, C: ConsensusStorage<N>> ExecutionState for BftExecutionState<N,
                 return Ok(None);
             }
 
+            // Timestamps are on the batches, not the transactions. We assume every batch contains
+            // at least one valid transaction.
+            let mut timestamps =
+                consensus_output.batches.iter().map(|batch| batch.0.metadata.created_at).collect::<Vec<_>>();
+            timestamps.sort();
+            // why u64, sui???
+            let median: i64 = timestamps[timestamps.len() / 2].try_into().unwrap();
```

Comment:
I don't condone `u64` for timestamps but the rationale is probably that they don't need the neg ones.

Reply:
yes as per the discussion in slack.